pkg/operator/ceph/cluster/controller.go

```diff
@@ -142,6 +143,24 @@ func (c *ClusterController) StartWatch(namespace string, stopCh chan struct{}) e
 	)
 	go nodeController.Run(stopCh)
 
+	// watch for updates to the device discovery configmap
+	_, deviceCMController := cache.NewInformer(
+		cache.NewFilteredListWatchFromClient(c.context.Clientset.CoreV1().RESTClient(),
+			"configmaps", namespace, func(options *metav1.ListOptions) {
```

Comment:
I believe the namespace we want to watch is the one where the operator and discover pod are running. 
- The `namespace` here is where the cluster CRD should be watched (either the operator's namespace, or blank, which means all namespaces)
- If needed, you can get the namespace from `os.Getenv(k8sutil.PodNamespaceEnvVar)`
- Using the operator's namespace should actually avoid the access denied errors you saw when watching all namespaces

Reply:
Hm, ok. I changed it to watch within `os.Getenv(k8sutil.PodNamespaceEnvVar)` which is the same thing used to create the discover daemonset. It'd be nice if the environment variable name had some baked-in semantic meaning (its currently the generic `POD_NAMESPACE`) as the current naming suggests that there isn't much equivalence between the operator and discover pod namespaces even if the usage enforces that.

Reply:
Yeah `POD_NAMESPACE` is a generic convention for the namespace of the currently running pod, which is always going to be the same for the operator and the discover pods.